commands/role.bee.inc

```diff
@@ -0,0 +1,386 @@
+        'module' => array(
+          'description' => bt("Get the permissions for this module."),
+          'value' => bt('Module'),
+        ),
+      ),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee permissions' => bt('Display a list of all permissions of the modules for the current site.'),
+        'bee permissions --module=node' => bt('Display a list of all permissions  from the module node for the current site.'),
+      ),
+    ),
+
+    'role-create' => array(
+      'description' => bt('Add a role.'),
+      'callback' => 'role_add_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'role' => bt('Role to add'),
+      ),
+      'aliases' => array('rcrt'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-create manager' => bt("Add role 'manager'."),
+      ),
+    ),
+    'role-delete' => array(
+      'description' => bt('Delete a role.'),
+      'callback' => 'role_delete_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'role' => bt('Role to delete'),
+      ),
+      'aliases' => array('rdel'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-delete manager' => bt("Delete role 'manager'."),
+      ),
+    ),
+    'role-add-perm' => array(
+      'description' => bt('Grant specified permission(s) to a role.'),
+      'callback' => 'role_add_permission_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'permissions' => bt('Permissions'),
+        'role' => bt('Role'),
+      ),
+      'aliases' => array('rap'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-add-perm  \'post comments\' \'anonymous user\'' => bt("Allow anon users to post comments."),
+        'bee role-add-perm  "\'view own unpublished content\' , \'view any unpublished content\' , \'view revisions\'" \'anonymous user\'' => bt("Grant multiple permissions to the anon users"),
+      ),
+    ),
+    'role-remove-perm' => array(
+      'description' => bt('Remove specified permission(s) from a role.'),
+      'callback' => 'role_remove_permission_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'permissions' => bt('Permissions'),
+        'role' => bt('Role'),
+      ),
+      'aliases' => array('rrp'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-remove-perm \'access content\' \'anonymous user\'' => bt("Hide content from anon users."),
+        'bee role-remove-perm  "\'view own unpublished content\' , \'view any unpublished content\' , \'view revisions\'" \'anonymous user\'' => bt("Remove multiple permissions from the anon users"),
+      ),
+    ),
+  );
+}
+
+/**
+ * Command callback: List all roles.
+ */
+function roles_bee_callback($arguments, $options) {
+  $rows = array();
+  $roles = user_roles(TRUE);
+  unset($roles['authenticated']);
```

Comment:
see PR #316?  your remark: 
Also, it would be good to:

    exclude authenticated - every user in this list will have it.

I followed that, i agree that also.

Reply:
Thanks for explaining where it came from; I searched through comments in this PR to see if I'd said it here, but didn't think about the earlier one.  I think this function is a different case; roles have permissions and they may well need to know what permissions authenticated has as other roles don't need to duplicate those permissions.  The purpose of the role is to see what permissions a role has, so I don't think there is a case to exclude it here.